Adding official allegro5 library

HerrNamenlos123 opened the pull request:
allegro5/5.2.7

I want to make the official allegro5 library available on Conan, so it's easier to use in my other projects and allows others to easily build my projects from source.

I am completely new to Conan, so please look over it and correct me whenever something's wrong. I did successfully test it on Windows and Ubuntu.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=9753) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=9753) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/conan-io/conan-center-index?pullRequest=9753) it.</sub>

HerrNamenlos123 commented:
I think I need help now. I added `libgl1-mesa-dev` as a dependency, because Allegro5 needs OpenGL. When I added the `opengl/system` conan package which installs `libopengl-dev` it was not enough, I had to manually add  `libgl1-mesa-dev` so that CMake would find it.

Apparently there is no such package for arm64 architecture, can anyone help me with this?

SSE4 commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,313 @@
+            self.requires("pulseaudio/14.2")
+            self.requires("gtk/system")
+            
+        self.options["freetype"].with_png = False
```
also, `requirements` method is not the best place doing that, `config_options`/`configure` fit better

SSE4 commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,313 @@
+
+        # Configure dependency flags for cmake
+        flags = "-Wno-dev"
+        flags += " -DSHARED=" + str(self.options.shared).lower()
```
it's preferred to use `CMake` helper and set definitions like:
```
self._cmake.definitions["WANT_DOCS"] = False
```

SSE4 commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,313 @@
+            flags += " -DPREFER_STATIC_DEPS=false"
+
+        # libpng dependency
+        tools.replace_in_file(str(os.path.join(self.build_folder, "allegro5/addons/image/CMakeLists.txt")), 
```
I think you can use `cmake_find_package[_multi]` generator in order to create find modules needed, instead of patching these lines here and below
but if it's absolutely needed and unavoidable, I think code could be unified and extracted, below there are too many similar blocks

SSE4 commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,313 @@
+
+    def build(self):
+        if self.settings.os == "Windows":
+            self.run("cd allegro5/build & cmake --build . --config RelWithDebInfo") # Build the project
```
it doesn't seem right to hard-code build type here. better to use `self.settings.build_type`

SpaceIm replied:
or just use CMake helper

SSE4 commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,313 @@
+            self.run("make") # Build the project
+
+    def package(self):
+        self.copy("*.h", dst="include", src="allegro5/include")
```
does it have an install target?

SSE4 commented on recipes/allegro5/all/test_package/al_ex_color2.c:
```diff
@@ -0,0 +1,367 @@
+#define ALLEGRO_UNSTABLE
+#include <allegro5/allegro.h>
```
consider reducing the example
it might be enough to just call a single function like `al_init_font_addon` to ensure example compile and links against the library

uilianries commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,313 @@
+class Allegro5Conan(ConanFile):
+    name = "allegro5"
+    version = "5.2.7"
+    license = "Custom licenses"
```
please, list the license names correctly: https://github.com/liballeg/allegro5/blob/master/LICENSE.txt

HerrNamenlos123 commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,313 @@
+
+class Allegro5Conan(ConanFile):
+    name = "allegro5"
+    version = "5.2.7"
```
What else should be there or how does the package know which version it is? Just delete this line?

HerrNamenlos123 replied:
Oh okay it's in config.yml

HerrNamenlos123 commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,313 @@
+            del self.options.fPIC
+
+    def source(self):
+        self.run("git clone https://github.com/liballeg/allegro5.git --depth=1 --single-branch --branch=5.2.7")
```
Yes, I did it like in the getting started docs

HerrNamenlos123 commented:
The hook tells me to add `required_conan_version >= "1.33.0"`, how can I do that or where must this go?

uilianries commented:
@HerrNamenlos123 Please, read https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#how-to-provide-a-good-recipe You can take a recipe from master branch as reference. 


Related to required_conan_version you can take a look on https://github.com/conan-io/conan-center-index/blob/cd325616aab3f9373a450f32805fbbcd50f901f3/recipes/pdqsort/all/conanfile.py

HerrNamenlos123 commented on recipes/allegro5/all/test_package/al_ex_color2.c:
```diff
@@ -0,0 +1,367 @@
+#define ALLEGRO_UNSTABLE
```
It was like that in the original example, i'm going to remove it. It just adds some useful functions AFAIK

HerrNamenlos123 replied:
I'm going to keep all four includes even though they are not used, to see if most headers are available

uilianries commented:
That's you can not force dependencies options, instead, prefer raising ConanInvalidConfiguration. Forcing dependencies will require a package ID which is not available as pre-built package, and Conan Center will not build it.

SpaceIm commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,327 @@
+            self.requires("libalsa/1.2.5.1")
+            self.requires("pulseaudio/14.2")
+            self.requires("gtk/system")
+            
+        self.options["freetype"].with_png = False
+        self.options["freetype"].with_zlib = False
+        self.options["freetype"].with_bzip2 = False
+        self.options["freetype"].with_brotli = False
+        self.options["opusfile"].http = False
```
```suggestion
```

HerrNamenlos123 replied:
I think i need help now. These lines are not the problem, otherwise the freetype test would not compile.

The problem is I wanted to remove all of the source patching and those lines above and I added `cmake_find_package` and `cmake_find_package_multi` generators, but some libraries are not recognized because the compile test fails and some other libraries are correctly found, but then allegro5 can't compile as some header files are not found.

For example when I add the generators and remove the patching, `zlib` is searched for multiple times and also found (and maybe skipped) multiple times and then allegro5 says `cannot find zlib.h`.

I really don't know what to do here. How can I make it that the libraries are found, by just adding the cmake_... generators?

HerrNamenlos123 replied:
This is my current file:

```
from conans import ConanFile, CMake, tools
from pathlib import Path
import os

required_conan_version = ">=1.33.0"

class Allegro5Conan(ConanFile):
    name = "allegro5"
    license = ("ZLib", "BSD", "SDL", "Dejavu-fonts", "Creative Commons BY Attribution", "CBString")
    homepage = "https://github.com/liballeg/allegro5"
    url = "https://github.com/conan-io/conan-center-index"
    description = "Cross-platform graphics framework for basic game development and desktop applications"
    topics = ("gamedev", "gui", "framework", "graphics")
    settings = "os", "compiler", "build_type", "arch"
    options = {"shared": [True, False], "fPIC": [True, False]}
    default_options = {"shared": False, "fPIC": True}
    
    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
    _cmake = None

    def requirements(self):       # Conditional dependencies
        self.requires("libpng/1.6.37")
        self.requires("zlib/1.2.11")
        self.requires("libjpeg/9d")
        self.requires("freetype/2.11.1")
        self.requires("libwebp/1.2.2")
        self.requires("flac/1.3.3")
        self.requires("ogg/1.3.5")
        self.requires("vorbis/1.3.7")
        self.requires("minimp3/20200304")
        self.requires("openal/1.21.1")
        self.requires("physfs/3.0.2")
        self.requires("opus/1.3.1")
        self.requires("opusfile/0.12")
        self.requires("theora/1.1.1")
        self.requires("opengl/system")

        if self.settings.os != "Windows":
            self.requires("xorg/system")
            self.requires("libalsa/1.2.5.1")
            self.requires("pulseaudio/14.2")
            self.requires("gtk/system")

    @property
    def _source_subfolder(self):
        return "source_subfolder"
        
    @property
    def _build_subfolder(self):
        return "build_subfolder"

    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC

    def configure(self):
        if self.options.shared:
            del self.options.fPIC
            
    def set_version(self):
        self.version = "5.2.7"

    def source(self):
        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)

    def _configure_cmake(self):
        if self._cmake:
            return self._cmake

        self._cmake = CMake(self)
        self._cmake.definitions["SHARED"] = self.options.shared
        self._cmake.definitions["CMAKE_BUILD_TYPE"] = self.settings.build_type
        self._cmake.definitions["WANT_DOCS"] = False
        self._cmake.definitions["WANT_DOCS_HTML"] = False
        self._cmake.definitions["WANT_EXAMPLES"] = False
        self._cmake.definitions["WANT_FONT"] = True
        self._cmake.definitions["WANT_MONOLITH"] = True
        self._cmake.definitions["WANT_TESTS"] = False
        self._cmake.definitions["WANT_DEMO"] = False
        self._cmake.definitions["WANT_RELEASE_LOGGING"] = False
        self._cmake.definitions["WANT_VORBIS"] = True
        self._cmake.definitions["WANT_MP3"] = True

        if self.settings.compiler == "Visual Studio":
            self._cmake.definitions["WANT_STATIC_RUNTIME"] = str(self.settings.compiler.runtime == "MT").lower()
            self._cmake.definitions["PREFER_STATIC_DEPS"] = True
        else:
            self._cmake.definitions["WANT_STATIC_RUNTIME"] = False
            self._cmake.definitions["PREFER_STATIC_DEPS"] = False
            
        # Disable specific compiler warnings
        if self.settings.compiler == "Visual Studio":   
            self._cmake.definitions["CMAKE_CXX_FLAGS"] = "/wd4267 /wd4018"    # possible loss of data warning, signed/unsigned comparison
        else:
            self._cmake.definitions["CMAKE_CXX_FLAGS"] = "-Wno-unused-variable -Wp,-w"  # unused variables, preprocessor warnings
        
        self._cmake.configure(source_folder=self._source_subfolder, build_folder=self._build_subfolder)
        return self._cmake

    def build(self):
        #self._patch_sources()
        cmake = self._configure_cmake()
        cmake.build()

    def package(self):
        self.copy("*", dst="include", src=os.path.join(self._source_subfolder, "include"), keep_path=True)
        self.copy("*.h", dst="include", src=os.path.join(self._build_subfolder, "include"), keep_path=True)
        self.copy("*.lib", dst="lib", src=self._build_subfolder, keep_path=False)
        self.copy("*.a", dst="lib", src=self._build_subfolder, keep_path=False)
        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder, keep_path=False)

        # Add all addon's headers
        for addon in os.listdir(os.path.join(self._source_subfolder, "addons")):
            self.copy("*.h", dst="include/allegro5", src=os.path.join(self._source_subfolder, "addons", addon, "allegro5"), keep_path=True)

    def package_info(self):
        self.cpp_info.libs = ["allegro_monolith-static"]

        if not self.options.shared:
            self.cpp_info.defines = ["ALLEGRO_STATICLINK"]

        if self.settings.os == "Windows":
            self.cpp_info.system_libs = [ "opengl32", "shlwapi" ]

```

It produces following console output:
```
[HOOK - conan-center.py] pre_export(): [DEPRECATED GLOBAL CPPSTD (KB-H001)] OK
[HOOK - conan-center.py] pre_export(): [REFERENCE LOWERCASE (KB-H002)] OK
[HOOK - conan-center.py] pre_export(): [RECIPE METADATA (KB-H003)] OK
[HOOK - conan-center.py] pre_export(): [HEADER_ONLY, NO COPY SOURCE (KB-H005)] OK
[HOOK - conan-center.py] pre_export(): [FPIC OPTION (KB-H006)] OK
[HOOK - conan-center.py] pre_export(): [VERSION RANGES (KB-H008)] OK
[HOOK - conan-center.py] pre_export(): [RECIPE FOLDER SIZE (KB-H009)] Total recipe size: 17.564453125 KB
[HOOK - conan-center.py] pre_export(): [RECIPE FOLDER SIZE (KB-H009)] OK
[HOOK - conan-center.py] pre_export(): [EXPORT LICENSE (KB-H023)] exports: None
[HOOK - conan-center.py] pre_export(): [EXPORT LICENSE (KB-H023)] exports: None
[HOOK - conan-center.py] pre_export(): [EXPORT LICENSE (KB-H023)] OK
[HOOK - conan-center.py] pre_export(): [TEST PACKAGE FOLDER (KB-H024)] OK
[HOOK - conan-center.py] pre_export(): [META LINES (KB-H025)] OK
[HOOK - conan-center.py] pre_export(): [CONAN CENTER INDEX URL (KB-H027)] OK
[HOOK - conan-center.py] pre_export(): [CMAKE MINIMUM VERSION (KB-H028)] OK
[HOOK - conan-center.py] pre_export(): [TEST PACKAGE - RUN ENVIRONMENT (KB-H029)] OK
[HOOK - conan-center.py] pre_export(): [SYSTEM REQUIREMENTS (KB-H032)] OK
[HOOK - conan-center.py] pre_export(): [CONANDATA.YML FORMAT (KB-H030)] OK
[HOOK - conan-center.py] pre_export(): [TEST PACKAGE - NO IMPORTS() (KB-H034)] OK
[HOOK - conan-center.py] pre_export(): [NO AUTHOR (KB-H037)] OK
[HOOK - conan-center.py] pre_export(): [NOT ALLOWED ATTRIBUTES (KB-H039)] OK
[HOOK - conan-center.py] pre_export(): [NO TARGET NAME (KB-H040)] OK
[HOOK - conan-center.py] pre_export(): [NO FINAL ENDLINE (KB-H041)] OK
[HOOK - conan-center.py] pre_export(): [NO REQUIRES.ADD() (KB-H044)] OK
[HOOK - conan-center.py] pre_export(): [DELETE OPTIONS (KB-H045)] OK
[HOOK - conan-center.py] pre_export(): [CMAKE VERBOSE MAKEFILE (KB-H046)] OK
[HOOK - conan-center.py] pre_export(): [CMAKE VERSION REQUIRED (KB-H048)] OK
[HOOK - conan-center.py] pre_export(): [CMAKE WINDOWS EXPORT ALL SYMBOLS (KB-H049)] OK
[HOOK - conan-center.py] pre_export(): [DEFAULT OPTIONS AS DICTIONARY (KB-H051)] OK
[HOOK - conan-center.py] pre_export(): [CONFIG.YML HAS NEW VERSION (KB-H052)] OK
[HOOK - conan-center.py] pre_export(): [PRIVATE IMPORTS (KB-H053)] OK
[HOOK - conan-center.py] pre_export(): [SINGLE REQUIRES (KB-H055)] OK
[HOOK - conan-center.py] pre_export(): [TOOLS RENAME (KB-H057)] OK
[HOOK - conan-center.py] pre_export(): [ILLEGAL CHARACTERS (KB-H058)] OK
[HOOK - conan-center.py] pre_export(): [CLASS NAME (KB-H059)] OK
[HOOK - conan-center.py] pre_export(): [NO CRLF (KB-H060)] OK
[HOOK - conan-center.py] pre_export(): [NO BUILD SYSTEM FUNCTIONS (KB-H061)] OK
[HOOK - conan-center.py] pre_export(): [TOOLS CROSS BUILDING (KB-H062)] OK
[HOOK - conan-center.py] pre_export(): [INVALID TOPICS (KB-H064)] OK
[HOOK - conan-center.py] pre_export(): [NO REQUIRED_CONAN_VERSION (KB-H065)] OK
[HOOK - conan-center.py] pre_export(): [TEST_TYPE MANAGEMENT (KB-H068)] OK
Exporting package recipe
allegro5/5.2.7 exports: File 'conandata.yml' found. Exporting it...
allegro5/5.2.7 exports: Copied 1 '.yml' file: conandata.yml
[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] Saving conandata.yml: {'sources': {'5.2.7': {'url': 'https://github.com/liballeg/allegro5/archive/refs/tags/5.2.7.0.tar.gz', 'sha256': '1ccc1804be54e6c84b58218cb63143d80d18bf6ed06a0ad49480a3616cddeee1'}}}
[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] New conandata.yml contents: sources:
  5.2.7:
    sha256: 1ccc1804be54e6c84b58218cb63143d80d18bf6ed06a0ad49480a3616cddeee1
    url: https://github.com/liballeg/allegro5/archive/refs/tags/5.2.7.0.tar.gz

[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] OK
[HOOK - conan-center.py] post_export(): [DEFAULT SHARED OPTION VALUE (KB-H050)] OK
[HOOK - conan-center.py] post_export(): [LICENSE PUBLIC DOMAIN (KB-H056)] OK
allegro5/5.2.7: A new conanfile.py version was exported
allegro5/5.2.7: Folder: C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\export
allegro5/5.2.7: Using the exported files summary hash as the recipe revision: 0284a281292e52565074f5abe8c5ee71 
allegro5/5.2.7: Package recipe modified in export, forcing source folder removal
allegro5/5.2.7: Use the --keep-source, -k option to skip it
allegro5/5.2.7: Exported revision: 0284a281292e52565074f5abe8c5ee71
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MT
compiler.version=17
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

WARN: theora/1.1.1: requirement ogg/1.3.4 overridden by allegro5/5.2.7 to ogg/1.3.5 
allegro5/5.2.7: Forced build from source
allegro5/5.2.7 (test package): Installing package
Requirements
    allegro5/5.2.7 from local cache - Cache
    brotli/1.0.9 from 'conancenter' - Cache
    bzip2/1.0.8 from 'conancenter' - Cache
    flac/1.3.3 from 'conancenter' - Cache
    freetype/2.11.1 from 'conancenter' - Cache
    libjpeg/9d from 'conancenter' - Cache
    libpng/1.6.37 from 'conancenter' - Cache
    libwebp/1.2.2 from 'conancenter' - Cache
    minimp3/20200304 from 'conancenter' - Cache
    ogg/1.3.5 from 'conancenter' - Cache
    openal/1.21.1 from 'conancenter' - Cache
    opengl/system from 'conancenter' - Cache
    openssl/1.1.1m from 'conancenter' - Cache
    opus/1.3.1 from 'conancenter' - Cache
    opusfile/0.12 from 'conancenter' - Cache
    physfs/3.0.2 from 'conancenter' - Cache
    theora/1.1.1 from 'conancenter' - Cache
    vorbis/1.3.7 from 'conancenter' - Cache
    zlib/1.2.11 from 'conancenter' - Cache
Packages
    allegro5/5.2.7:000cc2621df26bb6e9a3e09f0db3be8be6a75a14 - Build
    brotli/1.0.9:871c6303207aa1b539e682559d0b5180e15f0c30 - Cache
    bzip2/1.0.8:9ee7772e425d35c1aa48c680ed5911828505a114 - Cache
    flac/1.3.3:7d87384462cc7cba7e8871c04ab4cb443040b25d - Cache
    freetype/2.11.1:c5220703be4c5e0542d4ea80a456ea808c740e59 - Cache
    libjpeg/9d:871c6303207aa1b539e682559d0b5180e15f0c30 - Cache
    libpng/1.6.37:e6ef91f0424a68ca7da760e186d161ddb22122a4 - Cache
    libwebp/1.2.2:89ef5afe790d5dbccd88fbaab2f0a9cb95645f89 - Cache
    minimp3/20200304:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    ogg/1.3.5:871c6303207aa1b539e682559d0b5180e15f0c30 - Cache
    openal/1.21.1:871c6303207aa1b539e682559d0b5180e15f0c30 - Cache
    opengl/system:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    openssl/1.1.1m:871c6303207aa1b539e682559d0b5180e15f0c30 - Cache
    opus/1.3.1:871c6303207aa1b539e682559d0b5180e15f0c30 - Cache
    opusfile/0.12:fc60670ba712764ec04089a686cc1c1f6b1e826b - Cache
    physfs/3.0.2:33563738afb4743e0eaaecd7bd87dfa22fd6eab0 - Cache
    theora/1.1.1:7d87384462cc7cba7e8871c04ab4cb443040b25d - Cache
    vorbis/1.3.7:7d87384462cc7cba7e8871c04ab4cb443040b25d - Cache
    zlib/1.2.11:871c6303207aa1b539e682559d0b5180e15f0c30 - Cache

Installing (downloading, building) binaries...
brotli/1.0.9: Already installed!
bzip2/1.0.8: Already installed!
bzip2/1.0.8: Appending PATH environment variable: C:\Users\Florian\.conan\data\bzip2\1.0.8\_\_\package\9ee7772e425d35c1aa48c680ed5911828505a114\bin
libjpeg/9d: Already installed!
libwebp/1.2.2: Already installed!
minimp3/20200304: Already installed!
ogg/1.3.5: Already installed!
openal/1.21.1: Already installed!
opengl/system: Already installed!
openssl/1.1.1m: Already installed!
opus/1.3.1: Already installed!
physfs/3.0.2: Already installed!
zlib/1.2.11: Already installed!
flac/1.3.3: Already installed!
flac/1.3.3: Appending PATH environment variable: C:\Users\Florian\.conan\data\flac\1.3.3\_\_\package\7d87384462cc7cba7e8871c04ab4cb443040b25d\bin
libpng/1.6.37: Already installed!
opusfile/0.12: Already installed!
theora/1.1.1: Already installed!
vorbis/1.3.7: Already installed!
freetype/2.11.1: Already installed!
allegro5/5.2.7: WARN: Build folder is dirty, removing it: C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\build\000cc2621df26bb6e9a3e09f0db3be8be6a75a14
[HOOK - conan-center.py] pre_source(): [IMMUTABLE SOURCES (KB-H010)] OK
allegro5/5.2.7: Configuring sources in C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\source
Downloading 5.2.7.0.tar.gz completed [6935.48k]

[HOOK - conan-center.py] post_source(): [LIBCXX MANAGEMENT (KB-H011)] OK
[HOOK - conan-center.py] post_source(): [CPPSTD MANAGEMENT (KB-H022)] OK
[HOOK - conan-center.py] post_source(): [SHORT_PATHS USAGE (KB-H066)] OK
allegro5/5.2.7: Copying sources to build folder
allegro5/5.2.7: Building your package in C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\build\000cc2621df26bb6e9a3e09f0db3be8be6a75a14
allegro5/5.2.7: Generator cmake_find_package created FindJPEG.cmake
allegro5/5.2.7: Generator cmake_find_package created FindFreetype.cmake
allegro5/5.2.7: Generator cmake_find_package created FindWebP.cmake
allegro5/5.2.7: Generator cmake_find_package created Findflac.cmake
allegro5/5.2.7: Generator cmake_find_package created FindVorbis.cmake
allegro5/5.2.7: Generator cmake_find_package created Findminimp3.cmake
allegro5/5.2.7: Generator cmake_find_package created FindOpenAL.cmake
allegro5/5.2.7: Generator cmake_find_package created FindPhysFS.cmake
allegro5/5.2.7: Generator cmake_find_package created Findopusfile.cmake
allegro5/5.2.7: Generator cmake_find_package created Findtheora.cmake
allegro5/5.2.7: Generator cmake_find_package created Findopengl_system.cmake
allegro5/5.2.7: Generator cmake_find_package created FindPNG.cmake
allegro5/5.2.7: Generator cmake_find_package created FindBZip2.cmake
allegro5/5.2.7: Generator cmake_find_package created FindBrotli.cmake
allegro5/5.2.7: Generator cmake_find_package created FindOgg.cmake
allegro5/5.2.7: Generator cmake_find_package created FindOpus.cmake
allegro5/5.2.7: Generator cmake_find_package created FindOpenSSL.cmake
allegro5/5.2.7: Generator cmake_find_package created FindZLIB.cmake
allegro5/5.2.7: Generator cmake created conanbuildinfo.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created JPEGConfigVersion.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created JPEGConfig.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created JPEGTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created JPEGTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created freetype-config-version.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created freetype-config.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created freetypeTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created freetypeTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created WebPConfigVersion.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created WebPTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created WebPTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created WebPConfig.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created flac-config-version.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created flacTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created flacTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created flac-config.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created VorbisConfigVersion.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created VorbisTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created VorbisTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created VorbisConfig.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created minimp3-config-version.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created minimp3-config.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created minimp3Targets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created minimp3Target-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OpenALConfigVersion.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OpenALConfig.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OpenALTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OpenALTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created PhysFSConfigVersion.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created PhysFSConfig.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created PhysFSTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created PhysFSTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created opusfile-config-version.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created opusfileTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created opusfileTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created opusfile-config.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created theora-config-version.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created theoraTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created theoraTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created theora-config.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created opengl_system-config-version.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created opengl_system-config.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created opengl_systemTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created opengl_systemTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created PNGConfigVersion.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created PNGConfig.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created PNGTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created PNGTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created BZip2ConfigVersion.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created BZip2Config.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created BZip2Targets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created BZip2Target-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created BrotliConfigVersion.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created BrotliTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created BrotliTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created BrotliConfig.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OggConfigVersion.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OggTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OggTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OggConfig.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OpusConfigVersion.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OpusTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OpusTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OpusConfig.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OpenSSLConfigVersion.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OpenSSLTarget-release.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OpenSSLTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created OpenSSLConfig.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created ZLIBConfigVersion.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created ZLIBConfig.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created ZLIBTargets.cmake
allegro5/5.2.7: Generator cmake_find_package_multi created ZLIBTarget-release.cmake
allegro5/5.2.7: Aggregating env generators
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] 'fPIC' option not found
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK
allegro5/5.2.7: Calling build()
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19043.
-- The C compiler identification is MSVC 19.31.31104.0
-- The CXX compiler identification is MSVC 19.31.31104.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.31.31103/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.31.31103/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Guessed MSVC directory: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.31.31103/bin/Hostx64/x64/cl.exe
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Looking for include file dirent.h
-- Looking for include file dirent.h - not found
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Looking for include file linux/input.h
-- Looking for include file linux/input.h - not found
-- Looking for include file stdbool.h
-- Looking for include file stdbool.h - found
-- Looking for include file stdint.h
-- Looking for include file stdint.h - found
-- Looking for include file sys/io.h
-- Looking for include file sys/io.h - not found
-- Looking for include file sys/stat.h
-- Looking for include file sys/stat.h - found
-- Looking for include file sys/time.h
-- Looking for include file sys/time.h - not found
-- Looking for include file time.h
-- Looking for include file time.h - found
-- Looking for include file sys/utsname.h
-- Looking for include file sys/utsname.h - not found
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for include file soundcard.h
-- Looking for include file soundcard.h - not found
-- Looking for include file sys/soundcard.h
-- Looking for include file sys/soundcard.h - not found
-- Looking for include file machine/soundcard.h
-- Looking for include file machine/soundcard.h - not found
-- Looking for include file linux/soundcard.h
-- Looking for include file linux/soundcard.h - not found
-- Looking for include file libkern/OSAtomic.h
-- Looking for include file libkern/OSAtomic.h - not found
-- Looking for include file sys/inotify.h
-- Looking for include file sys/inotify.h - not found
-- Looking for include file sal.h
-- Looking for include file sal.h - found
-- Looking for getexecname
-- Looking for getexecname - not found
-- Looking for mkstemp
-- Looking for mkstemp - not found
-- Looking for mmap
-- Looking for mmap - not found
-- Looking for mprotect
-- Looking for mprotect - not found
-- Looking for sched_yield
-- Looking for sched_yield - not found
-- Looking for sysconf
-- Looking for sysconf - not found
-- Looking for fseeko
-- Looking for fseeko - not found
-- Looking for ftello
-- Looking for ftello - not found
-- Looking for strerror_r
-- Looking for strerror_r - not found
-- Looking for strerror_s
-- Looking for strerror_s - found
-- Looking for _ftelli64
-- Looking for _ftelli64 - found
-- Looking for _fseeki64
-- Looking for _fseeki64 - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of _Bool
-- Check size of _Bool - done
-- Performing Test ALLEGRO_HAVE_PROCFS_ARGCV
-- Performing Test ALLEGRO_HAVE_PROCFS_ARGCV - Failed
-- Performing Test ALLEGRO_HAVE_SV_PROCFS_H
-- Performing Test ALLEGRO_HAVE_SV_PROCFS_H - Failed
-- Performing Test ALLEGRO_HAVE_VA_COPY
-- Performing Test ALLEGRO_HAVE_VA_COPY - Success
-- Found OpenGL: opengl32   
Using OpenGL
-- Found DINPUT: C:/Program Files (x86)/Microsoft DirectX SDK (June 2010)/Include  
-- Found D3D9: C:/Program Files (x86)/Microsoft DirectX SDK (June 2010)/Include  
-- Found D3DX9: C:/Program Files (x86)/Microsoft DirectX SDK (June 2010)/Include  
-- Found DSOUND: C:/Program Files (x86)/Microsoft DirectX SDK (June 2010)/Include  
-- Found XINPUT: C:/Program Files (x86)/Microsoft DirectX SDK (June 2010)/Include  
-- Could NOT find GDIPLUS (missing: GDIPLUS_INCLUDE_DIR) 
WARNING: cannot use GDI+. Will try other libraries.
WARNING: FreeImage not found, disabling support
-- Conan: Using autogenerated FindPNG.cmake
-- Found PNG: 1.6.37 (found version "1.6.37") 
-- Library libpng16 found C:/Users/Florian/.conan/data/libpng/1.6.37/_/_/package/e6ef91f0424a68ca7da760e186d161ddb22122a4/lib/libpng16.lib
-- Found: C:/Users/Florian/.conan/data/libpng/1.6.37/_/_/package/e6ef91f0424a68ca7da760e186d161ddb22122a4/lib/libpng16.lib
-- Conan: Using autogenerated FindZLIB.cmake
-- Found ZLIB: 1.2.11 (found version "1.2.11") 
-- Library zlib found C:/Users/Florian/.conan/data/zlib/1.2.11/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/zlib.lib
-- Found: C:/Users/Florian/.conan/data/zlib/1.2.11/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/zlib.lib
-- Conan: Using autogenerated FindJPEG.cmake
-- Found JPEG: 9d (found version "9d") 
-- Library libjpeg found C:/Users/Florian/.conan/data/libjpeg/9d/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/libjpeg.lib
-- Found: C:/Users/Florian/.conan/data/libjpeg/9d/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/libjpeg.lib
-- Conan: Using autogenerated FindWebP.cmake
-- Found WebP: 1.2.2 (found version "1.2.2")
-- Library webpdecoder found C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webpdecoder.lib
-- Found: C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webpdecoder.lib
-- Library webpdemux found C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webpdemux.lib
-- Found: C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webpdemux.lib
-- Library webpmux found C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webpmux.lib
-- Found: C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webpmux.lib
-- Library webp found C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webp.lib
-- Found: C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webp.lib
-- Library webp found C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webp.lib
-- Found: C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webp.lib
-- Library webpmux found C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webpmux.lib
-- Found: C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webpmux.lib
-- Library webpdemux found C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webpdemux.lib
-- Found: C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webpdemux.lib
-- Library webpdecoder found C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webpdecoder.lib
-- Found: C:/Users/Florian/.conan/data/libwebp/1.2.2/_/_/package/89ef5afe790d5dbccd88fbaab2f0a9cb95645f89/lib/webpdecoder.lib
-- Conan: Using autogenerated FindOpenAL.cmake
-- Found OpenAL: 1.21.1 (found version "1.21.1") 
-- Library OpenAL32 found C:/Users/Florian/.conan/data/openal/1.21.1/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/OpenAL32.lib
-- Found: C:/Users/Florian/.conan/data/openal/1.21.1/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/OpenAL32.lib
-- Could NOT find OPENSL (missing: OPENSL_INCLUDE_DIR OPENSL_LIBRARY) 
-- Conan: Using autogenerated Findflac.cmake
-- Found flac: 1.3.3 (found version "1.3.3") 
-- Library FLAC++ found C:/Users/Florian/.conan/data/flac/1.3.3/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/FLAC++.lib
-- Found: C:/Users/Florian/.conan/data/flac/1.3.3/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/FLAC++.lib
-- Library FLAC found C:/Users/Florian/.conan/data/flac/1.3.3/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/FLAC.lib
-- Found: C:/Users/Florian/.conan/data/flac/1.3.3/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/FLAC.lib
-- Conan: Using autogenerated FindOgg.cmake
-- Found Ogg: 1.3.5 (found version "1.3.5")
-- Library ogg found C:/Users/Florian/.conan/data/ogg/1.3.5/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/ogg.lib
-- Found: C:/Users/Florian/.conan/data/ogg/1.3.5/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/ogg.lib
-- Library ogg found C:/Users/Florian/.conan/data/ogg/1.3.5/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/ogg.lib
-- Found: C:/Users/Florian/.conan/data/ogg/1.3.5/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/ogg.lib
-- Library FLAC found C:/Users/Florian/.conan/data/flac/1.3.3/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/FLAC.lib
-- Found: C:/Users/Florian/.conan/data/flac/1.3.3/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/FLAC.lib
-- Library FLAC++ found C:/Users/Florian/.conan/data/flac/1.3.3/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/FLAC++.lib
-- Found: C:/Users/Florian/.conan/data/flac/1.3.3/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/FLAC++.lib
-- Performing Test FLAC_COMPILES
-- Performing Test FLAC_COMPILES - Failed
WARNING: libFLAC not found or compile test failed, disabling support.
-- Could NOT find DUMB (missing: DUMB_INCLUDE_DIR DUMB_LIBRARY) 
WARNING: libdumb >= 2.0 or <= 0.9.3 not found or compile test failed, disabling support. See <https://github.com/kode54/dumb> for 2.0 or <http://dumb.sourceforge.net/> for 0.9.3.
-- Conan: Using autogenerated FindVorbis.cmake
-- Found Vorbis: 1.3.7 (found version "1.3.7")
-- Library vorbisenc found C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisenc.lib
-- Found: C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisenc.lib
-- Library vorbisfile found C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisfile.lib
-- Found: C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisfile.lib
-- Library vorbis found C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbis.lib
-- Found: C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbis.lib
-- Conan: Dependency Ogg already found
-- Library vorbis found C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbis.lib
-- Found: C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbis.lib
-- Library vorbisfile found C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisfile.lib
-- Found: C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisfile.lib
-- Library vorbisenc found C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisenc.lib
-- Found: C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisenc.lib
-- Performing Test VORBIS_COMPILES
-- Performing Test VORBIS_COMPILES - Failed
WARNING: libvorbis not found or compile test failed, disabling support.
-- Conan: Using autogenerated FindOpus.cmake
-- Found Opus: 1.3.1 (found version "1.3.1")
-- Library opus found C:/Users/Florian/.conan/data/opus/1.3.1/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/opus.lib
-- Found: C:/Users/Florian/.conan/data/opus/1.3.1/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/opus.lib
-- Library opus found C:/Users/Florian/.conan/data/opus/1.3.1/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/opus.lib
-- Found: C:/Users/Florian/.conan/data/opus/1.3.1/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/opus.lib
-- Performing Test OPUS_COMPILES
-- Performing Test OPUS_COMPILES - Failed
WARNING: libopus not found or compile test failed, disabling support.
-- Conan: Using autogenerated Findminimp3.cmake
-- Found minimp3: 20200304 (found version "20200304")
-- Conan: Using autogenerated FindFreetype.cmake
-- Found Freetype: 2.11.1 (found version "2.11.1") 
-- Library freetype found C:/Users/Florian/.conan/data/freetype/2.11.1/_/_/package/c5220703be4c5e0542d4ea80a456ea808c740e59/lib/freetype.lib
-- Found: C:/Users/Florian/.conan/data/freetype/2.11.1/_/_/package/c5220703be4c5e0542d4ea80a456ea808c740e59/lib/freetype.lib
-- Conan: Using autogenerated FindPNG.cmake
-- Library libpng16 found C:/Users/Florian/.conan/data/libpng/1.6.37/_/_/package/e6ef91f0424a68ca7da760e186d161ddb22122a4/lib/libpng16.lib
-- Found: C:/Users/Florian/.conan/data/libpng/1.6.37/_/_/package/e6ef91f0424a68ca7da760e186d161ddb22122a4/lib/libpng16.lib
-- Conan: Using autogenerated FindZLIB.cmake
-- Library zlib found C:/Users/Florian/.conan/data/zlib/1.2.11/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/zlib.lib
-- Found: C:/Users/Florian/.conan/data/zlib/1.2.11/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/zlib.lib
-- Dependency ZLIB already found
-- Conan: Using autogenerated FindBZip2.cmake
-- Found BZip2: 1.0.8 (found version "1.0.8")
-- Library bz2 found C:/Users/Florian/.conan/data/bzip2/1.0.8/_/_/package/9ee7772e425d35c1aa48c680ed5911828505a114/lib/bz2.lib
-- Found: C:/Users/Florian/.conan/data/bzip2/1.0.8/_/_/package/9ee7772e425d35c1aa48c680ed5911828505a114/lib/bz2.lib
-- Conan: Using autogenerated FindBrotli.cmake
-- Found Brotli: 1.0.9 (found version "1.0.9") 
-- Library brotlidec-static found C:/Users/Florian/.conan/data/brotli/1.0.9/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/brotlidec-static.lib
-- Found: C:/Users/Florian/.conan/data/brotli/1.0.9/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/brotlidec-static.lib
-- Library brotlienc-static found C:/Users/Florian/.conan/data/brotli/1.0.9/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/brotlienc-static.lib
-- Found: C:/Users/Florian/.conan/data/brotli/1.0.9/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/brotlienc-static.lib
-- Library brotlicommon-static found C:/Users/Florian/.conan/data/brotli/1.0.9/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/brotlicommon-static.lib
-- Found: C:/Users/Florian/.conan/data/brotli/1.0.9/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/brotlicommon-static.lib
-- Library brotlicommon-static found C:/Users/Florian/.conan/data/brotli/1.0.9/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/brotlicommon-static.lib
-- Found: C:/Users/Florian/.conan/data/brotli/1.0.9/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/brotlicommon-static.lib
-- Library brotlienc-static found C:/Users/Florian/.conan/data/brotli/1.0.9/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/brotlienc-static.lib
-- Found: C:/Users/Florian/.conan/data/brotli/1.0.9/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/brotlienc-static.lib
-- Library brotlidec-static found C:/Users/Florian/.conan/data/brotli/1.0.9/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/brotlidec-static.lib
-- Found: C:/Users/Florian/.conan/data/brotli/1.0.9/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/brotlidec-static.lib
-- Performing Test TTF_COMPILES
-- Performing Test TTF_COMPILES - Success
-- Conan: Using autogenerated FindPhysFS.cmake
-- Found PhysFS: 3.0.2 (found version "3.0.2") 
-- Library physfs-static found C:/Users/Florian/.conan/data/physfs/3.0.2/_/_/package/33563738afb4743e0eaaecd7bd87dfa22fd6eab0/lib/physfs-static.lib
-- Found: C:/Users/Florian/.conan/data/physfs/3.0.2/_/_/package/33563738afb4743e0eaaecd7bd87dfa22fd6eab0/lib/physfs-static.lib
-- Could NOT find PHYSFS (missing: PHYSFS_LIBRARY PHYSFS_INCLUDE_DIR)
-- Conan: Using autogenerated FindZLIB.cmake
-- Library zlib found C:/Users/Florian/.conan/data/zlib/1.2.11/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/zlib.lib
-- Skipping already existing target: CONAN_LIB::ZLIB_zlib
-- Found: C:/Users/Florian/.conan/data/zlib/1.2.11/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/zlib.lib
-- Conan: Using autogenerated Findtheora.cmake
-- Found theora: 1.1.1 (found version "1.1.1") 
-- Library libtheora_static found C:/Users/Florian/.conan/data/theora/1.1.1/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/libtheora_static.lib
-- Found: C:/Users/Florian/.conan/data/theora/1.1.1/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/libtheora_static.lib
-- Conan: Using autogenerated FindOgg.cmake
-- Library ogg found C:/Users/Florian/.conan/data/ogg/1.3.5/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/ogg.lib
-- Found: C:/Users/Florian/.conan/data/ogg/1.3.5/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/ogg.lib
-- Library ogg found C:/Users/Florian/.conan/data/ogg/1.3.5/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/ogg.lib
-- Found: C:/Users/Florian/.conan/data/ogg/1.3.5/_/_/package/871c6303207aa1b539e682559d0b5180e15f0c30/lib/ogg.lib
-- Library libtheora_static found C:/Users/Florian/.conan/data/theora/1.1.1/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/libtheora_static.lib
-- Found: C:/Users/Florian/.conan/data/theora/1.1.1/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/libtheora_static.lib
-- Conan: Using autogenerated FindVorbis.cmake
-- Library vorbisenc found C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisenc.lib
-- Found: C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisenc.lib
-- Library vorbisfile found C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisfile.lib
-- Found: C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisfile.lib
-- Library vorbis found C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbis.lib
-- Found: C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbis.lib
-- Conan: Dependency Ogg already found
-- Library vorbis found C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbis.lib
-- Found: C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbis.lib
-- Library vorbisfile found C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisfile.lib
-- Found: C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisfile.lib
-- Library vorbisenc found C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisenc.lib
-- Found: C:/Users/Florian/.conan/data/vorbis/1.3.7/_/_/package/7d87384462cc7cba7e8871c04ab4cb443040b25d/lib/vorbisenc.lib
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DATAROOTDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CONAN_COMPILER
    CONAN_COMPILER_VERSION
    CONAN_CXX_FLAGS
    CONAN_C_FLAGS
    CONAN_EXPORTED
    CONAN_IN_LOCAL_CACHE
    CONAN_LINK_RUNTIME


-- Build files have been written to: C:/Users/Florian/.conan/data/allegro5/5.2.7/_/_/build/000cc2621df26bb6e9a3e09f0db3be8be6a75a14/build_subfolder
Microsoft (R)-Build-Engine, Version 17.1.0+ae57d105c für .NET Framework
Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.

  Checking Build System
  Building Custom Rule C:/Users/Florian/.conan/data/allegro5/5.2.7/_/_/build/000cc2621df26bb6e9a3e09f0db3be8be6a75a14/source_subfolder/CMakeLists.txt
  allegro.c
  bitmap.c
  bitmap_draw.c
  bitmap_io.c
  bitmap_lock.c
  bitmap_pixel.c
  bitmap_type.c
  blenders.c
  clipboard.c
  config.c
  convert.c
  cpu.c
  debug.c
  display.c
  display_settings.c
  drawing.c
  dtor.c
  events.c
  evtsrc.c
  exitfunc.c
  file.c
  file_slice.c
  file_stdio.c
  fshook.c
  fshook_stdio.c
  fullscreen_mode.c
  haptic.c
  inline.c
  joynu.c
  keybdnu.c
  libc.c
  math.c
  memblit.c
  memdraw.c
  memory.c
  monitor.c
  mousenu.c
  mouse_cursor.c
  path.c
  pixels.c
  shader.c
  system.c
  threads.c
  timernu.c
  tls.c
  touch_input.c
  transformations.c
  tri_soft.c
  utf8.c
  aatree.c
  bstrlib.c
  list.c
  vector.c
  wclipboard.c
  whapall.c
  whapdrv.c
  whapxi.c
  wjoyall.c
  wjoydrv.c
  wjoyxi.c
  wkeyboard.c
  wmcursor.c
  wmouse.c
  wsystem.c
  wthread.c
  wtime.c
  wtouch_input.c
  wunicode.c
  wwindow.c
  wxthread.c
  extensions.c
  ogl_bitmap.c
  C:\Program Files (x86)\Microsoft DirectX SDK (June 2010)\Include\dinput.h: DIRECTINPUT_VERSION undefined. Defaulting to version 0x0800
  ogl_display.c
  ogl_draw.c
  ogl_fbo.c
  ogl_lock.c
  ogl_lock_es.c
  ogl_render_state.c
  ogl_shader.c
  wgl_disp.c
  high_primitives.c
  line_soft.c
  point_soft.c
  polygon.c
  polyline.c
  prim_opengl.c
  prim_soft.c
  prim_util.c
  primitives.c
  triangulator.c
  bmp.c
  iio.c
  pcx.c
  tga.c
  dds.c
  identify.c
  png.c
C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\build\000cc2621df26bb6e9a3e09f0db3be8be6a75a14\source_subfolder\addons\image\png.c(7,10): fatal error C1083: Cannot open include file: 'zlib.h'
: No such file or directory [C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\build\000cc2621df26bb6e9a3e09f0db3be8be6a75a14\build_subfolder\allegro_monolith.vcxproj]
  jpg.c
  webp.c
  font.c
C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\build\000cc2621df26bb6e9a3e09f0db3be8be6a75a14\source_subfolder\addons\image\webp.c(6,10): fatal error C1083: Cannot open include file: 'webp/d
ecode.h': No such file or directory [C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\build\000cc2621df26bb6e9a3e09f0db3be8be6a75a14\build_subfolder\allegro_monolith.vcxproj]
  fontbmp.c
  stdfont.c
  text.c
  bmfont.c
  xml.c
  audio.c
  audio_io.c
  kcm_dtor.c
  kcm_instance.c
  kcm_mixer.c
  kcm_sample.c
  kcm_stream.c
  kcm_voice.c
  recorder.c
  openal.c
  acodec.c
  wav.c
  helper.c
  voc.c
  mp3.c
  ttf.c
  color.c
C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\build\000cc2621df26bb6e9a3e09f0db3be8be6a75a14\source_subfolder\addons\acodec\mp3.c(19,10): fatal error C1083: Cannot open include file: 'minim
p3.h': No such file or directory [C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\build\000cc2621df26bb6e9a3e09f0db3be8be6a75a14\build_subfolder\allegro_monolith.vcxproj]
  memfile.c
  dialog.c
  textlog.c
  menu.c
  win_dialog.c
  video.c
  ogv.c
  generic_main.c
C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\build\000cc2621df26bb6e9a3e09f0db3be8be6a75a14\source_subfolder\addons\video\ogv.c(23,10): fatal error C1083: Cannot open include file: 'ogg/og
g.h': No such file or directory [C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\build\000cc2621df26bb6e9a3e09f0db3be8be6a75a14\build_subfolder\allegro_monolith.vcxproj]
allegro5/5.2.7: 
allegro5/5.2.7: ERROR: Package '000cc2621df26bb6e9a3e09f0db3be8be6a75a14' build failed
allegro5/5.2.7: WARN: Build folder C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\build\000cc2621df26bb6e9a3e09f0db3be8be6a75a14
ERROR: allegro5/5.2.7: Error in build() method, line 301
        cmake.build()
        ConanException: Error 1 while executing cmake --build C:\Users\Florian\.conan\data\allegro5\5.2.7\_\_\build\000cc2621df26bb6e9a3e09f0db3be8be6a75a14\build_subfolder --config Release -- /m:4 /verbosity:minimal
```

SpaceIm replied:
> I think i need help now. These lines are not the problem, otherwise the freetype test would not compile.

These lines prevent to find a pre-built package of freetype in conan-center.

> I really don't know what to do here. How can I make it that the libraries are found, by just adding the cmake_... generators?

No magic, analyse upstream CMakeLists to understand what's wrong.

SpaceIm commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,164 @@
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["SHARED"] = False
+        self._cmake.definitions["WANT_STATIC_RUNTIME"] = "MT" in self.settings.compiler.runtime if self._is_msvc else False
```
```suggestion
        self._cmake.definitions["WANT_STATIC_RUNTIME"] = "MT" in msvc_runtime_flag(self) if self._is_msvc else False
```

HerrNamenlos123 commented:
Fiiiinallyyyyy!!!

Does everything look okay now?

SpaceIm commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,164 @@
+        self._cmake.definitions["WANT_MP3"] = True
+        self._cmake.definitions["WANT_OGG_VIDEO"] = True
+
+        self._cmake.definitions["FREETYPE_ZLIB"] = True
+        self._cmake.definitions["FREETYPE_BZIP2"] = True
+        self._cmake.definitions["FREETYPE_PNG"] = True
```
```suggestion
        self._cmake.definitions["FREETYPE_ZLIB"] = self.options["freetype"].with_zlib
        self._cmake.definitions["FREETYPE_BZIP2"] = self.options["freetype"].with_bzip2
        self._cmake.definitions["FREETYPE_PNG"] = self.options["freetype"].with_png
```

Not sure it's really required. You could set these values to False I guess. allegro5 CMakeLists is really ugly and fragile (old style CMake without relying on imported targets).

HerrNamenlos123 replied:
Yeah I noticed, allegro's CMake is horrible. Glad it finally compiles now on all platforms, I'm still going to accept this.

SpaceIm reviewed (changes requested):
Few suggestions

HerrNamenlos123 commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,164 @@
+    description = "Cross-platform graphics framework for basic game development and desktop applications"
+    topics = ("allegro5", "gamedev", "gui", "framework", "graphics")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "fPIC": True
+    }
```
Is the shared option really required?

HerrNamenlos123 replied:
I tried adding it, but weirdly this is the result: `/usr/bin/ld: cannot find -lZLIB::ZLIB`

SpaceIm replied:
> Is the shared option really required?

Yes, it's. better

> I tried adding it, but weirdly this is the result: `/usr/bin/ld: cannot find -lZLIB::ZLIB` (while linking the allegro library)

`ZLIB::ZLIB` is the CMake imported target of zlib. Again, you have to dig in their CMakeLists to understand what kind of weird things they are doing.

HerrNamenlos123 replied:
When the allegro library is shared, should all the dependencies like zlib also be shared?

uilianries replied:
No. They are static by default, but you still can force all shared and installing by passing the option `-o *:shared`

HerrNamenlos123 replied:
I would really rather not do that. I spent a long time trying around, but I can't figure out where this zlib error is coming from.

HerrNamenlos123 commented:
```
self._cmake.definitions["CMAKE_BUILD_TYPE"] = self.settings.build_type
```
This line is necessary, I added it back in. Without it, msvc wouldn't compile

uilianries commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,165 @@
+        self._cmake.definitions["WANT_TESTS"] = False
+        self._cmake.definitions["WANT_DEMO"] = False
+        self._cmake.definitions["WANT_RELEASE_LOGGING"] = False
+        self._cmake.definitions["WANT_VORBIS"] = True
```
I see that vorbis and some other dependencies are optional. Why not make them as optional on the recipe as well?

uilianries commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,165 @@
+            
+        self._cmake.definitions["CMAKE_CXX_FLAGS"] = "/wd4267 /wd4018" if self._is_msvc else "-Wno-unused-variable -w"
+        
+        self._cmake.configure(source_folder=self._source_subfolder, build_folder=self._build_subfolder)
```
```suggestion
        self._cmake.configure(build_folder=self._build_subfolder)
```

Need to use cmake wrapper. As example: https://github.com/conan-io/conan-center-index/blob/master/recipes/aaplus/all/CMakeLists.txt

uilianries commented on recipes/allegro5/all/test_package/CMakeLists.txt:
```diff
@@ -0,0 +1,11 @@
+project(test_package C)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+message("Dir: ${CMAKE_BINARY_DIR}")
```
```suggestion
```

uilianries commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,165 @@
+
+class Allegro5Conan(ConanFile):
+    name = "allegro5"
+    license = ("ZLib", "BSD", "SDL", "Dejavu-fonts", "Creative Commons BY Attribution", "CBString")
```
```suggestion
    license = ("ZLib", "BSD-3-Clause")
```
Reading https://github.com/liballeg/allegro5/blob/master/LICENSE.txt

I can found [Zlib](https://spdx.org/licenses/Zlib.html) and [BSD-3-Clause](https://spdx.org/licenses/BSD-3-Clause.html)

Examples and demos we not packaging, neither redistributing.

HerrNamenlos123 replied:
Oh, right

SpaceIm commented:
> All green in build 28 (`41382ff87a86c251c1eeb55e84b38a66d0ad1d16`):
> 
> * **allegro5/5.2.7@:**
>   All packages built successfully! ([All logs](https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/9753/28/summary.json))

`INVALID_CONFIGURATION` for gcc 5 and all versions of apple-clang due to freeimage. https://github.com/conan-io/conan-center-index/pull/9930 will fix this issue, and therefore this PR will be able to test macOS.

SpaceIm commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,165 @@
+        self.requires("theora/1.1.1")
+        self.requires("opengl/system")
+
+        if self.settings.os != "Windows":
+            self.requires("xorg/system")
```
xorg recipe raises if not Linux or FreeBSD.

SpaceIm commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,165 @@
+            self.requires("glu/system")
+            self.requires("libalsa/1.2.5.1")
+            self.requires("pulseaudio/14.2")
+            self.requires("gtk/system")
```
why not using the proper recipe of gtk? system recipes are the last resort.

SpaceIm commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -0,0 +1,165 @@
+        if self.settings.os != "Windows":
+            self.requires("xorg/system")
+            self.requires("glu/system")
+            self.requires("libalsa/1.2.5.1")
+            self.requires("pulseaudio/14.2")
```
libalsa & pulsaudio are Linux only

SpaceIm commented on recipes/allegro5/all/conanfile.py:
```diff
@@ -19,7 +19,7 @@ class Allegro5Conan(ConanFile):
     }
     
-    generators = "cmake", "cmake_find_package"
+    generators = "cmake", "cmake_find_package", "pkg_config"
```
pkg_config in generators means that you have to add pkgconf in build requirements.
